test/src/tests-identity.ts

```diff
@@ -72,97 +77,194 @@ const BadCallbackAsString = ('badCallbackString' as unknown) as Callback;
+            let data: IIdentityAPIRequestData;
+
+            waitForCondition(hasIdentifyReturned)
+                .then(() => {
+                    // 4 Calls should be expected
+                    // 1. Identify
+                    // 2. Session Start
+                    // 3. AST
+                    expect(fetchMock.calls().length).to.equal(3);
+
+                    const firstCall = fetchMock.calls()[0];
+                    expect(firstCall[0].split('/')[4]).to.equal('identify');
+
+                    data = JSON.parse(
+                        firstCall[1].body as unknown as string
+                    ) as IIdentityAPIRequestData;
+
+                    expect(data).to.have.keys(
+                        'client_sdk',
+                        'environment',
+                        'known_identities',
+                        'previous_mpid',
+                        'request_id',
+                        'request_timestamp_ms',
+                        'context'
+                    );
 
-            mParticle.Identity.identify({
-                userIdentities: {
-                    email: 'test@email.com',
+                    expect(data.previous_mpid).to.equal(null);
+
+                    expect(data.known_identities).to.have.property(
+                        'device_application_stamp'
+                    );
+
+                    done();
+                })
+                .catch(done);
+        });
+
+        it('should contain previous mpid in a repeated identify request', function (done) {
+            fetchMockSuccess(urls.identify, {
+                context: null,
+                matched_identities: {
+                    device_application_stamp: 'my-das',
                 },
+                is_ephemeral: true,
+                // This should be a new mpid, since we identified earlier
+                // in the beforeEach method
+                mpid: 'new-mpid',
+                is_logged_in: false,
             });
-            const data = getIdentityEvent(mockServer.requests, 'identify');
 
-            data.should.have.properties(
-                'client_sdk',
-                'environment',
-                'known_identities',
-                'previous_mpid',
-                'request_id',
-                'request_timestamp_ms',
-                'context'
-            );
+            let data: IIdentityAPIRequestData;
 
-            expect(data.previous_mpid).to.equal(null);
+            waitForCondition(hasIdentifyReturned)
+                .then(() => {
+                    mParticle.Identity.identify({
+                        userIdentities: {
+                            email: 'test@email.com',
+                        },
+                    });
+
+                    // 4 Calls should be expected
+                    // 1. Identify
+                    // 2. Session Start
+                    // 3. AST
+                    // 4. Identify
```

Comment:
In general, we should be doing a `waitForCondition(hasIdentifyReturned)` if `mParticle.init` happens inside of the `beforeEach`, then in this case, another identify call is happening, and the waitForCondition(secondHasIdentifyReturned) should _follow_ the `identify` call, not precede it.

The reason this test passes is because it is only testing what is being sent in the second identify call, as opposed to having to wait for a response before doing something else (like logging an event), that is then being tested.

See here for an example. https://github.com/mParticle/mparticle-web-sdk/pull/921/files#diff-78454482c0c860ae432945ed8b1d45433ba0f3a1c95258c65c93c4b95cd9c17cR1033-R1049